<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2025-10-30 Thu 10:26 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>A Rigorous, Interpretable Genetic Algorithm Framework Reveals Statistically Significant Gene Signatures in Parkinson's Disease Single-Cell Data</title>
<meta name="author" content="Sam Neaves" />
<meta name="generator" content="Org Mode" />
<style>
  #content { max-width: 60em; margin: auto; }
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #e6e6e6;
    border-radius: 3px;
    background-color: #f2f2f2;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
  }
  pre.src:before {
    display: none;
    position: absolute;
    top: -8px;
    right: 12px;
    padding: 3px;
    color: #555;
    background-color: #f2f2f299;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-authinfo::before { content: 'Authinfo'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { }
</style>
<script>
  window.MathJax = {
    tex: {
      ams: {
        multlineWidth: '85%'
      },
      tags: 'ams',
      tagSide: 'right',
      tagIndent: '.8em'
    },
    chtml: {
      scale: 1.0,
      displayAlign: 'center',
      displayIndent: '0em'
    },
    svg: {
      scale: 1.0,
      displayAlign: 'center',
      displayIndent: '0em'
    },
    output: {
      font: 'mathjax-modern',
      displayOverflow: 'overflow'
    }
  };
</script>

<script
  id="MathJax-script"
  async
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
</script>
</head>
<body>
<div id="content" class="content">
<h1 class="title">A Rigorous, Interpretable Genetic Algorithm Framework Reveals Statistically Significant Gene Signatures in Parkinson's Disease Single-Cell Data</h1>
<div id="table-of-contents" role="doc-toc">
<h2>Table of Contents</h2>
<div id="text-table-of-contents" role="doc-toc">
<ul>
<li><a href="#orgcf16eed">1. Abstract</a></li>
<li><a href="#org9d02c4e">2. Introduction</a></li>
<li><a href="#org10ea020">3. Methods</a>
<ul>
<li><a href="#org57cfec1">3.1. Single-Cell Data Preprocessing:</a></li>
<li><a href="#orgf2007a1">3.2. The Genetic Algorithm Framework:</a>
<ul>
<li><a href="#orgd9f449d">3.2.1. </a></li>
<li><a href="#orgddfc856">3.2.2. A Diverse Subgroup Set Discovery (DSSD) Fitness Function</a></li>
</ul>
</li>
<li><a href="#orga652b7d">3.3. Statistical Validation: The "Best-of-N-Repeats" Permutation Test:</a></li>
</ul>
</li>
<li><a href="#org04ad2c1">4. HPC Implementation:</a></li>
<li><a href="#orgd09fad6">5. Results</a></li>
<li><a href="#orge197e55">6. Discussion</a></li>
<li><a href="#orgd5f4ef5">7. Conclusion:</a></li>
</ul>
</div>
</div>

<div id="outline-container-orgcf16eed" class="outline-2">
<h2 id="orgcf16eed"><span class="section-number-2">1.</span> Abstract</h2>
<div class="outline-text-2" id="text-1">
<ul class="org-ul">
<li>Background:
Briefly state the challenge: Parkinson's Disease (PD) is
heterogeneous, and single-cell (scRNA-seq) data offers a window into
this, but discovering the complex, combinatorial gene patterns that
define disease subgroups is difficult. Standard methods (like
differential expression) are too simple, while "black-box" AI models
(like deep learning) are not interpretable by biologists.</li>
</ul>


<ul class="org-ul">
<li>Methodology:
We applied an interpretable subgroup discovery framework using a
Genetic Algorithm (GA) to a high-dimensional scRNA-seq dataset of
dopaminergic neurons from PD patients and controls. Our method
evolves sets of human-readable, IF-THEN logical rules to identify
disease-associated subgroups.</li>
</ul>


<ul class="org-ul">
<li>Rigor:
To validate our findings and ensure they were not statistical
artifacts from searching a high-dimensional space, we performed a
massive-scale permutation test (1000 permutations). This involved
over 200,000 independent GA runs and consumed ~10,000 GPU-hours on
the DAWN supercomputer.</li>
</ul>


<ul class="org-ul">
<li>Results:
The GA identified a novel, non-obvious rule set (a combination of
gene expression states) that was highly statistically significant
(empirical p &lt; 0.001). This rule set achieved a fitness on the real
data that was not matched by any of the 999 permuted (randomized)
datasets. The resulting rules point to a combinatorial signature
involving genes such as [Gene A, Gene B, Gene C&#x2026;].</li>
</ul>


<ul class="org-ul">
<li>Conclusion:
This work provides a statistically-validated, interpretable, and
biologically relevant hypothesis for PD pathogenesis. It also
demonstrates a powerful, generalizable framework for applying
interpretable AI with large-scale HPC to extract robust, testable
insights from complex single-cell data.</li>
</ul>
</div>
</div>


<div id="outline-container-org9d02c4e" class="outline-2">
<h2 id="org9d02c4e"><span class="section-number-2">2.</span> Introduction</h2>
<div class="outline-text-2" id="text-2">
<ul class="org-ul">
<li>The Biological Problem:
Start with Parkinson's Disease. It's a complex, devastating
neurodegenerative disease. Its progression and presentation are
highly heterogeneous, suggesting multiple underlying mechanisms or
patient subgroups.</li>
</ul>


<ul class="org-ul">
<li>The Data Opportunity:
The emergence of single-cell RNA sequencing (scRNA-seq) allows us,
for the first time, to probe this heterogeneity at the resolution of
individual cells, like the vulnerable dopaminergic neurons.</li>
</ul>


<ul class="org-ul">
<li><p>
The "Interpretability Gap" (Your Justification):
This is your key argument.
</p>

<p>
State that simple statistical methods (like differential gene
expression) are useful but limited, as they only find individual
gene changes and miss the combinatorial logic of biological
pathways.
</p></li>
</ul>


<p>
State that "black-box" machine learning (e.g., Deep Learning, SVMs)
can be highly predictive but fail to provide interpretable
insights. A 95% accurate model is useless to a biologist if it
cannot explain why it made its decision.
</p>


<p>
This creates a need for methods that are both powerful (can search
high-dimensional, combinatorial spaces) and interpretable (produce
human-readable, testable hypotheses).
</p>


<ul class="org-ul">
<li>Our Framework (GA + Rigor):
Introduce your solution.
Propose the use of a rule-based Genetic Algorithm, a class of
evolutionary algorithm well-suited for complex combinatorial
optimization. Our framework is specifically designed for Diverse
Subgroup Set Discovery (DSSD). We hypothesize that Parkinson's is
not a monolithic disease but a heterogeneous condition with
multiple, distinct dysfunctional cell states. Therefore, unlike
standard methods that find a single 'best' rule, our GA evolves a
set of rules, where the entire set is evaluated for quality and
diversity. Each chromosome in our GA represents a model of this
heterogeneity, with each rule potentially identifying a distinct
biological subgroup. To validate these models, we paired the GA with a massive-scale permutation</li>
</ul>
<p>
test&#x2026;"
</p>

<p>
"Critically, we implemented this GA in PyTorch to leverage GPU parallelization, transforming the computationally infeasible task of a 1000-permutation test into a tractable one."
</p>

<ul class="org-ul">
<li>Justify the GA:
Explicitly state it was chosen because it produces interpretable IF-THEN rules that represent gene combinations.</li>

<li>Justify the Rigor:
State that to overcome the risk of "p-hacking" or finding spurious
correlations in the high-dimensional data, you paired the GA with a
massive-scale permutation test.</li>
</ul>


<ul class="org-ul">
<li>Punchline:
In this paper, we apply this rigorous, interpretable framework to a
[Your Data] scRNA-seq dataset. We leverage the DAWN HPC system to
perform ~10,000 GPU-hours of computation, resulting in the discovery
of a statistically significant (p &lt; 0.001) and interpretable gene
signature for a PD subgroup.</li>
</ul>
</div>
</div>

<div id="outline-container-org10ea020" class="outline-2">
<h2 id="org10ea020"><span class="section-number-2">3.</span> Methods</h2>
<div class="outline-text-2" id="text-3">
<p>
This section must be crystal clear and reproducible.
</p>
</div>

<div id="outline-container-org57cfec1" class="outline-3">
<h3 id="org57cfec1"><span class="section-number-3">3.1.</span> Single-Cell Data Preprocessing:</h3>
<div class="outline-text-3" id="text-3-1">
<ul class="org-ul">
<li>Source: Describe the SNatlas<sub>DaNs</sub><sub>seurat.RData</sub> object.</li>

<li>Filtering: Detail the cell filtering (e.g., keeping only CTR and PD labels PD<sub>B5</sub>-6, PD<sub>B3</sub>-4, resulting in 6,385 cells).</li>

<li>Confounder Correction: State that you used SCTransform to normalize data and simultaneously regress out the donor batch effect (the Sample<sub>v2</sub> variable). This is crucial for rigor.</li>

<li>Feature Selection: Explain that you performed a DE test on the corrected data and selected the top 1000 most variant genes for the GA, balancing computational feasibility with discovery.</li>

<li>Binarization: Describe the z-scoring method ((expression - mean) / std<sub>dev</sub>) and binarization (&gt; 0 -&gt; 1, &lt;= 0 -&gt; 0) to create the final 6385 x 1000 input matrix.</li>
</ul>
</div>
</div>

<div id="outline-container-orgf2007a1" class="outline-3">
<h3 id="orgf2007a1"><span class="section-number-3">3.2.</span> The Genetic Algorithm Framework:</h3>
<div class="outline-text-3" id="text-3-2">
</div>
<div id="outline-container-orgd9f449d" class="outline-4">
<h4 id="orgd9f449d"><span class="section-number-4">3.2.1.</span> </h4>
<div class="outline-text-4" id="text-3-2-1">
<ul class="org-ul">
<li>Chromosome Representation: Explain that each "individual" is a chromosome representing a set of k rules (where k was 3-6). Explain the "mask + value" structure for each rule (e.g., 2 * 1000 = 2000 bits per rule).</li>

<li>Fitness Function: This is critical. Define Fitness = Quality / Overlap.</li>

<li>Explain Quality = WRAcc(Union(Rules)). Justify WRAcc as a metric that balances subgroup size (support) and accuracy (lift).</li>

<li>Explain Overlap = TotalMatches / UniqueMatches as a penalty to encourage diverse, non-redundant rule sets.</li>

<li>GA Parameters: State them clearly: population<sub>size</sub>=55,
num<sub>generations</sub>=1000, tournament selection (size=5), single-point
crossover, bit-flip mutation (rate=0.0001), elitism (count=2).</li>
</ul>
</div>
</div>

<div id="outline-container-orgddfc856" class="outline-4">
<h4 id="orgddfc856"><span class="section-number-4">3.2.2.</span> A Diverse Subgroup Set Discovery (DSSD) Fitness Function</h4>
<div class="outline-text-4" id="text-3-2-2">
<p>
Our primary goal was not to find the single <b>best</b> rule describing PD,
as this would ignore the known heterogeneity of the disease. Standard
subgroup discovery algorithms often suffer from redundancy, where the
top 10 "best" rules are trivial variations of each other, all
describing the same, largest subgroup.
</p>

<p>
To overcome this, we designed our framework for <b>Diverse Subgroup Set
Discovery (DSSD)</b>. The central hypothesis is that our GA chromosome,
as a set of rules:
</p>

<p>
\[
\{ \text{Rule}_1, \text{Rule}_2, \ldots, \text{Rule}_k \}
\]
</p>

<p>
can serve as an interpretable model of distinct cell states.
</p>

<p>
To force the GA to find such a set, we designed a custom,
set-based fitness function that simultaneously rewards both quality
(exploitation) and diversity (exploration):
</p>

<p>
\[
\text{Fitness} =
\frac{\text{Quality}\big(\text{Union}(\text{Rules})\big)}
     {\text{Overlap}(\text{Rules})}
\]
</p>

<ol class="org-ol">
<li><p>
<b>Quality (WRAcc of the Union)</b>  
The quality of a chromosome is measured by the Weighted Relative Accuracy (WRAcc)
of the union of all cells covered by its rules.
</p>

<p>
\[
   \text{WRAcc} = P(G) \times \big(P(\text{Class} \mid G) - P(\text{Class})\big)
   \]
</p>

<p>
where \( G \) is the subgroup.  
This metric balances <b>support</b> (the size of the total subgroup found, \( P(G) \))
with <b>lift</b> (the enrichment of PD cells in that subgroup,
\( P(\text{Class} \mid G) - P(\text{Class}) \)).
It rewards chromosomes that identify a large and highly enriched population of PD cells.
</p></li>

<li><p>
<b>Diversity (Overlap Penalty)</b>  
The denominator, <b>Overlap</b>, is a penalty for redundancy. It is defined as:
</p>

<p>
\[
   \text{Overlap} = \frac{\text{TotalMatches}}{\text{UniqueMatches}}
   \]
</p>

<p>
where <b>UniqueMatches</b> is the number of cells covered by the union (as in the quality metric),
and <b>TotalMatches</b> is the sum of cells covered by each rule individually.
If all rules in a set cover the exact same cells, the Overlap score will be high (equal to \( k \)),
heavily penalizing the chromosome’s fitness.
</p></li>
</ol>

<p>
<b>Interpretation:</b>  
This fitness function creates an evolutionary pressure that punishes redundant rule sets
and rewards sets where each rule identifies a different population of cells,
all while maintaining high overall enrichment for the PD phenotype.
A high-fitness chromosome is therefore interpreted as a model of distinct,
disease-associated cell states.
</p>

<p>
<b>*</b>
</p>

<p>
2.c. Scalable Implementation: A GPU-Accelerated Genetic Algorithm
While the DSSD fitness function defines what to find, a computationally efficient search algorithm is required to find it. Standard subgroup discovery implementations (e.g., in R or sequential Python) are often iterative, CPU-bound, and computationally slow. A single run on a dataset of this scale can take many minutes to hours.
</p>

<p>
This slowness makes a rigorous permutation test—which requires running the entire discovery process 1,000+ times—computationally infeasible.
</p>

<p>
To solve this, we adapted the DSSD framework to a Genetic Algorithm (GA), as the population-based nature of GAs is "embarrassingly parallel." We specifically designed our GA implementation to leverage massive GPU parallelization using PyTorch and the Intel Extension for PyTorch (IPEX).
</p>

<p>
The key adaptation was vectorizing the entire fitness calculation. Instead of iterating through each chromosome one by one, we:
</p>

<p>
Represented the entire population for a batch of permutations as a single large tensor [N, P, C], where N is the number of permutations (e.g., 16), P is the population size (e.g., 55), and C is the chromosome length.
</p>

<p>
Implemented the get<sub>rule</sub><sub>matches</sub> function as a series of large-scale tensor broadcast operations. This creates a massive intermediate tensor ([N, P, R, S, F]) that allows the GPU to check every rule, for every chromosome, for every permutation, against every sample, for every feature simultaneously.
</p>

<p>
All subsequent calculations (WRAcc, overlap) were also performed as parallel tensor operations, resulting in a fitness score for every chromosome in the batch.
</p>

<p>
This GPU-accelerated design is the critical enabler of our study. It collapses the runtime for a single, complex DSSD run (e.g., 1000 generations, pop=55) from hours on a CPU to approximately 2-3 minutes on a single Intel Max 1550 GPU.
</p>

<p>
This massive speedup is what made our "best-of-N-repeats" permutation test, which consumed ~10,000 GPU-hours in total, computationally feasible. It allowed us to apply the same intense search effort (200 repeats * 1000 generations) to both the real data and all 999 permuted datasets, ensuring statistical rigor.
</p>
</div>
</div>
</div>

<div id="outline-container-orga652b7d" class="outline-3">
<h3 id="orga652b7d"><span class="section-number-3">3.3.</span> Statistical Validation: The "Best-of-N-Repeats" Permutation Test:</h3>
<div class="outline-text-3" id="text-3-3">
<p>
This is your most impressive method. Explain it in detail.
</p>

<ul class="org-ul">
<li>Null Hypothesis: That the observed gene expression patterns are not associated with the PD/CTR labels.</li>

<li>Procedure: To test this, you created 1000 datasets (1 real + 999 permuted by shuffling labels).</li>

<li>Rigor Justification: To ensure a fair comparison, the exact same computational effort was applied to every single dataset.</li>

<li>Effort per Dataset: Define the effort: "Each of the 1000 datasets was searched by N=200 independent GA runs, each starting from a different random population. The single highest-fitness chromosome found across all 200 repeats was recorded as the 'best' result for that dataset."</li>

<li>P-Value Calculation: The empirical p-value is (count(F<sub>perm</sub><sub>i</sub> &gt;= F<sub>real</sub>) + 1) / (999 + 1).</li>
</ul>
</div>
</div>
</div>

<div id="outline-container-org04ad2c1" class="outline-2">
<h2 id="org04ad2c1"><span class="section-number-2">4.</span> HPC Implementation:</h2>
<div class="outline-text-2" id="text-4">
<ul class="org-ul">
<li>Briefly state the GA was implemented in Python using PyTorch/IPEX for GPU acceleration.</li>

<li>Explain the experiment was executed as a 1000-task SLURM job array (&#x2013;array=0-999) on the DAWN supercomputer.</li>

<li>State the resource request (&#x2013;time=10:00:00, &#x2013;gres=gpu:1) and the
total cost: "This comprehensive, statistically rigorous search
consumed approximately 10,000 GPU-hours."</li>
</ul>
</div>
</div>



<div id="outline-container-orgd09fad6" class="outline-2">
<h2 id="orgd09fad6"><span class="section-number-2">5.</span> Results</h2>
<div class="outline-text-2" id="text-5">
<ol class="org-ol">
<li>A Statistically Significant Signature is Discovered:</li>

<li>State the main finding: "The 'best-of-200' search on the real data (Task 0) yielded a rule set with a fitness of [F<sub>real</sub>]."</li>

<li>"The distribution of 'best-of-200' fitness scores from the 999 permuted datasets had a mean of [Avg<sub>F</sub><sub>perm</sub>] and a maximum of [Max<sub>F</sub><sub>perm</sub>]."</li>

<li>"No permuted dataset achieved a fitness as high as the real data. This yields an empirical p-value of p &lt; 0.001 (0 observed / 999 permutations, +1/+1).</li>

<li>THE FIGURE: You must include a histogram here. It will show the beautiful bell curve of the 999 permuted fitness scores, with a single red line for your F<sub>real</sub> score sitting far to the right, all alone. This figure is the central proof of your paper.</li>

<li>The Interpretable PD Signature:</li>

<li>Present the winning rule set (the Rank 1 rules from the ga<sub>results</sub><sub>0.txt</sub> file).</li>

<li>List the rules clearly:</li>

<li>Rule 1: TPPP=0 AND SEC62=0 AND COX6C=0 &#x2026;</li>

<li><p>
Rule 2: NLGN4Y=0 AND COL5A1=0 &#x2026;
</p>

<p>
&#x2026;
</p></li>

<li>Subgroup Characterization (The so what):</li>
</ol>

<p>
Report the support (e.g., "This rule set identifies a subgroup of [X] cells&#x2026;") and precision (e.g., "&#x2026;of which [Y]% are PD cells, representing a [Z]-fold enrichment over baseline.")
</p>

<p>
(Optional, but powerful) Show a UMAP plot of all 6,385 cells, with the
cells identified by your rule set colored differently. Do they
cluster?
</p>
</div>
</div>

<div id="outline-container-orge197e55" class="outline-2">
<h2 id="orge197e55"><span class="section-number-2">6.</span> Discussion</h2>
<div class="outline-text-2" id="text-6">
<ol class="org-ol">
<li>Summary of Findings: Re-state the main finding (a p &lt; 0.001 interpretable signature was found) and emphasize that its strength comes from the combination of genes, not any single gene.</li>

<li>Biological Interpretation (Your collaboration): This is where you
discuss the meaning of the genes in the rules. "The appearance of
TPPP, COX6C, and [Gene X] in the top rules is highly
suggestive&#x2026; TPPP is involved in&#x2026; COX6C is a component of&#x2026; The
fact that our model found a rule combining TPPP=0 (low) and COX6C=0
(low) suggests a combinatorial failure in [X pathway]&#x2026; This is a
novel, testable hypothesis for a specific PD subgroup."</li>

<li><p>
The DSSD Framework Successfully Modeled Heterogeneity:
Our goal was not just to find a single signature, but to model PD's
heterogeneity. The DSSD framework proved highly effective. The
top-ranking chromosome (p &lt; 0.001) consisted of 6 distinct rules
(see Results, Table X). An analysis of these rules shows they are
not redundant; they identify different, though potentially
overlapping, subsets of the cell population.
</p>

<p>
For example, Rule 1 (TPPP=0 AND COX6C=0 &#x2026;) points to a subgroup
defined by [X genes, e.g., mitochondrial function], while Rule 3
(COL16A1=0 AND SDC3=0 &#x2026;) identifies a different subgroup defined
by [Y genes, e.g., extracellular matrix]. This provides a
data-driven model of multiple potential dysfunctional states
co-existing within the PD patient population. These distinct
rule-defined subgroups can now be analyzed independently (e.g., via
pathway analysis or visualization on the UMAP) to explore these
different facets of the disease. This demonstrates the power of
DSSD over simpler methods that would have missed this richness.
</p>

<p>
A key contribution of this work is a scalable framework for
rigorous DSSD. By adapting the DSSD concept to a fully vectorized,
GPU-accelerated Genetic Algorithm, we demonstrated that large-scale
permutation testing (consuming 10,000+ GPU-hours) is not only
possible but necessary for validating findings in high-dimensional
data. This HPC-centric approach provides a blueprint for future
studies aiming to bridge the gap between interpretable AI and
statistical robust ai.
</p></li>
</ol>


<ol class="org-ol">
<li>Methodological Reflections (Your "Experience"):</li>

<li>On Interpretability: Reiterate that this result is immediately useful to biologists, unlike a black-box model.</li>

<li>On Rigor: Discuss the necessity of the 10,000-hour permutation test. Show the Avg<sub>F</sub><sub>perm</sub> was non-zero, proving that even random data can produce seemingly good rules. This justifies the need for the massive computation to prove your real result was not random.</li>

<li>On HPC: State that this level of statistical rigor, while computationally expensive, is now feasible on modern HPC platforms like DAWN and should be considered a new standard for AI-driven discovery in biomedicine.</li>

<li>Limitations: Be upfront.</li>

<li>The search was limited to 1000 pre-selected genes.</li>

<li>The binarization of data is a simplification.</li>

<li>The results are a computational hypothesis, not a wet-lab-validated fact (this is the next paper).</li>
</ol>
</div>
</div>

<div id="outline-container-orgd5f4ef5" class="outline-2">
<h2 id="orgd5f4ef5"><span class="section-number-2">7.</span> Conclusion:</h2>
<div class="outline-text-2" id="text-7">
<p>
"We have successfully applied an interpretable, scalable, and
statistically rigorous GA framework to discover a novel gene
signature in PD. This work provides a concrete, data-driven
hypothesis for PD researchers and serves as a powerful case study
for the use of interpretable AI in complex biomedical discovery."
</p>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="date">Date: 2025-10-30 Thu 00:00</p>
<p class="author">Author: Sam Neaves</p>
<p class="date">Created: 2025-10-30 Thu 10:26</p>
<p class="validation"><a href="https://validator.w3.org/check?uri=referer">Validate</a></p>
</div>
</body>
</html>
